pkgs/misc/dxvk/default.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, pkgs
+, stdenv
+, fetchFromGitHub
+, pkgsCross
+  # Make it easy for users to build different versions of DXVK if those have better compatability
+  # with their games.
```

Comment:
Yeah sure but why is the file split? We can achieve everything done here in one file. And overriding src is way easier than providing an input for version and hash.

Reply:
I’ll refactor it to use the self pattern and add comments explaining why it’s doing what it’s doing, but DXVK needs to be two derivations. The core derivation builds Windows DLLs. The one that most users will use is for Unix and builds the Windows derivation with the correct cross compilers. It provides a setup script (to set up a Wine prefix for DXVK) and the DLLs in a `bin` output.

Reply:
Yes, please add that as a comment. Now it makes more sense why you choose to do it like this.

Reply:
Latest commits include an explanation:

>  # DXVK needs to be a separate derivation because it’s actually a set of DLLs for Windows that
>  # needs to be built with a cross-compiler.